libs/application/template-api-modules/src/lib/modules/templates/social-insurance-administration/social-insurance-administration.service.ts

```diff
@@ -0,0 +1,267 @@
+        .promise()
+      const fileContent = file.Body as Buffer
+
+      if (!fileContent) {
+        throw new Error('File content was undefined')
+      }
+
+      return fileContent.toString('base64')
+    } catch (e) {
+      this.logger.error('Cannot get ' + fileUpload + ' attachment', { e })
+      throw new Error('Failed to get the ' + fileUpload + ' attachment')
+    }
+  }
+
+  async sendApplication({ application, auth }: TemplateApiModuleActionProps) {
+    try {
+      if (application.typeId === ApplicationTypes.OLD_AGE_PENSION) {
+        const attachments = await this.getAttachments(application)
+
+        const oldAgePensionDTO = transformApplicationToOldAgePensionDTO(
+          application,
+          attachments,
+        )
+
+        const applicationType = getApplicationType(application).toLowerCase()
+
+        const response = await this.siaClientService.sendApplication(
+          auth,
+          oldAgePensionDTO,
+          applicationType,
+        )
+        return response
+      }
+    } catch (e) {
+      this.logger.error('Failed to send the old age pension application', e)
+      throw this.parseErrors(e)
+    }
+  }
+
+  async getApplicant({ auth }: TemplateApiModuleActionProps) {
+    const res = await this.siaClientService.getApplicant(auth).catch(() => {
+      throw new TemplateApiError(coreErrorMessages.defaultTemplateApiError, 500)
+    })
+
+    // mock data since gervimenn don't have bank account registered at TR,
+    // and might also not have phone number and email address registered
+    if (isRunningOnEnvironment('local')) {
+      if (res.bankAccount) {
+        res.bankAccount.bank = '2222'
+        res.bankAccount.ledger = '00'
+        res.bankAccount.accountNumber = '123456'
+
+        // ;(res.bankAccount.iban = 'RU02 0445 2560 0407 0281 0412 3456 7890 1'),
+        //   (res.bankAccount.swift = 'BARC GB 22 XXX'),
+        //   (res.bankAccount.foreignBankAddress = 'mosagata'),
+        //   (res.bankAccount.foreignBankName = 'banki'),
+        //   (res.bankAccount.currency = 'AUD')
+      }
+
+      if (!res.emailAddress) {
+        res.emailAddress = 'mail@mail.is'
+      }
+
+      if (!res.phoneNumber) {
+        res.phoneNumber = '888-8888'
+      }
+    }
+
+    if (!res.emailAddress) {
+      throw new TemplateApiError(
+        {
+          title: errorMessages.noEmailFound,
+          summary: errorMessages.noEmailFoundDescription,
+        },
+        500,
+      )
+    }
+
+    return res
+  }
+
+  async getIsEligible({ application, auth }: TemplateApiModuleActionProps) {
+    try {
+      if (isRunningOnEnvironment('local')) {
+        return { isEligible: true }
+      }
+      const applicationType = getApplicationType(application).toLowerCase()
+      return await this.siaClientService.getIsEligible(auth, applicationType)
+    } catch (e) {
+      throw new TemplateApiError(coreErrorMessages.defaultTemplateApiError, 500)
```

Comment:
The template api Catches all errors thrown and logs them. This would throw a templat api error and the actual error trace would not be logged. You can just remove the try catch.